examples/common/java/src/main/java/org/eea/tcf/worker/chain/services/WorkOrderRegistryService.java

```diff
@@ -0,0 +1,155 @@
+
+        if (registryAddress != null && !registryAddress.isEmpty()) {
+            try {
+                return WorkOrderRegistry.load(
+                        registryAddress, web3j, credentialsService.getCredentials(), contractGasProvider);
+            } catch (EnsResolutionException e) {
+                throw exceptionInInitializerError;
+            }
+        } else {
+            throw exceptionInInitializerError;
+        }
+    }
+
+    public WorkOrderRegistry getWorkOrderRegistryContract() {
+        return getWorkOrderRegistryContract(new DefaultGasProvider());
+    }
+
+
+    Flowable<WorkOrderSubmittedEvent> getWorkOrderEventObservableToLatest(BigInteger from) {
+        return getWorkOrderEventObservable(from, null);
+    }
+
+    Flowable<WorkOrderSubmittedEvent> getWorkOrderEventObservable(BigInteger from, BigInteger to) {
+        DefaultBlockParameter fromBlock = DefaultBlockParameter.valueOf(from);
+        DefaultBlockParameter toBlock = null;
+
+        if (to != null)
+            toBlock = DefaultBlockParameter.valueOf(to);
+
+        return getWorkOrderRegistryContract().workOrderSubmittedEventFlowable(fromBlock, toBlock)
+                .filter(workOrderSubmittedEventResponse -> {
+                    // If the WorkOrder matches the workerId we registered, it should only be 20 bytes long
+                    byte[] myWorkerId = stringToBytes(WorkOrderRegistryService.this.workerId);
+                    return compareBytes(workOrderSubmittedEventResponse.workerID, myWorkerId, 20, 12, 0);
+                })
+                .map(workOrderNotice -> {
+                    return WorkOrderSubmittedEvent.builder()
+                            .workOrderId(bytesToString(workOrderNotice.workOrderID))
+                            .workerId(bytesToString(workOrderNotice.workerID))
+                            .requesterId(bytesToString(workOrderNotice.requesterID))
+                            .build();
+                });
+    }
+
+    public boolean workOrderComplete(String workOrderId,
+                                     int returnCode,
+                                     String response,
+                                     String signature) {
+        byte[] workOrderIdBytes = stringToBytes32(workOrderId);
+        byte[] signatureByte = stringToBytes(signature);
+
+        TransactionReceipt completeReceipt;
+        RemoteCall<TransactionReceipt> completeCall = getWorkOrderRegistryContract().workOrderComplete(
+                        workOrderIdBytes, BigInteger.valueOf(returnCode), response, signatureByte);
+
+        try {
+            completeReceipt = completeCall.send();
+            log.error("Work order complete [workOrderId:{}, returnCode:{}, gasUsed:{}]",
+                    workOrderId, returnCode, completeReceipt.getGasUsed());
+        } catch (Exception e) {
+            log.error("Failed to call WorkOrderComplete [workOrderId:{}, returnCode:{}, exeption:{}]",
```

Comment:
From PR #37
s/exeption/exception/

Reply:
This one slipped my mind, I fixed it in a new commit.